### PR TITLE
Testing code for reset.R and unit-test for reset functions

### PR DESCRIPTION
**Pull request by kar-agg-gen**

### CODE REVIEW

File: tests/testthat/test_reset.R [hunk 1]
```diff
@@ -2,5 +2,40 @@ context("DGEobj - tests for reset.R functions")
...
+                 fixed  = TRUE)
+
+    # checking DGEobj without platformType
+    test_DGEobj <- setAttributes(item = DGEobj, list("level" = "gene"))
+    test_DGEobj <- setAttributes(item = test_DGEobj, list("PlatformType" = NULL))
+    expect_error(resetDGEobj(test_DGEobj),
+                 regexp = "Required attribute \"PlatformType\" is missing.",
+                 fixed  = TRUE)
+
+    # checking DGEobj with wrong counts matirx
```
Comment (saranya-ag) [inline tests/testthat/test_reset.R]: Rephrase comment to Testing DGEobj without count Matrix. 
Typo - Matrix

File: tests/testthat/test_reset.R [hunk 2]
```diff
@@ -2,5 +2,40 @@ context("DGEobj - tests for reset.R functions")
...
+    test_DGEobj <- setAttributes(item = test_DGEobj, list("PlatformType" = "RNA-Seq"))
+    expect_error(resetDGEobj(test_DGEobj),
+                 regexp = "`%in%`(x = itemName, table = names(dgeObj)) is not TRUE",
+                 fixed  = TRUE)
+
+    # checking DGEobj with wrong data
+    names(test_DGEobj) <- c( "counts_orig" , "counts", "design_orig" , "design", "peptideAnnotation_orig")
```
Comment (saranya-ag) [inline tests/testthat/test_reset.R]: Extra space before counts_orig

File: tests/testthat/test_reset.R [hunk 3]
```diff
@@ -2,5 +2,40 @@ context("DGEobj - tests for reset.R functions")
...
+    expect_error(resetDGEobj(test_DGEobj),
+                 regexp = "Gene/isoform/exon/protein data not found",
+                 fixed  = TRUE)
+
+    # checking valid object
+    names(test_DGEobj) <- c( "counts_orig" , "counts", "design_orig" , "design", "isoformData_orig")
```
Comment (saranya-ag) [inline tests/testthat/test_reset.R]: Extra space before counts_orig

File: tests/testthat/test_reset.R [hunk 4]
```diff
@@ -2,5 +2,40 @@ context("DGEobj - tests for reset.R functions")
...
+    # Expecting warning as current object could not build the GRanges object!
```
Comment (saranya-ag) [inline tests/testthat/test_reset.R]: Rephrase comment 

Warning expected as the current object cannot build the GRanges object.

File: R/reset.R [hunk 5]
```diff
@@ -26,24 +24,21 @@
...
         platformType <- tolower(attr(dgeObj, "PlatformType"))
+    } else {
+        stop("Required attribute \"PlatformType\" is missing.")
+    }
 
-    if (tolower(platformType) %in% platform.rnaseq)
-        counts <- getItem(dgeObj, "counts_orig")
-
-    design <- getItem(dgeObj, "design_orig")
+    metaList <- getBaseType(dgeObj, "meta")[1:3]
```
Comment (saranya-ag) [inline R/reset.R]: Can you check this - MetaList is not being used anywhere

**Review by saranya-ag — CHANGES_REQUESTED** [review]

**Review by saranya-ag — APPROVED** [review]

**Comment by kar-agg-gen** [comment]
closing PR as changes are already merged

**Pull request closed without merging**